packages/cms/pages/resources/resource/resource.tsx

```diff
@@ -189,10 +189,26 @@ export const FilterList: React.FunctionComponent = () => {
 export const Table: React.FunctionComponent = () => {
   const [pageSize, setPageSize] = useState(10)
   const [pageIndex, setPageIndex] = useState(0)
+
   const [selectedItems, setSelectedItems] = useState<any[]>([])
   const [sortField, setSortField] = useState('firstName')
   const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>('asc')
-  const { resource, applyFilter } = useResourceStore()
+  const { resource, applyFilter, fetchTableData } = useResourceStore()
+  const [items, setItems] = useState([])
+
+  useEffect(() => {
+    const getData = async () => {
+      const [{ data }, error] = await fetchTableData(
+        resource?.slugPlural.toLowerCase()!,
+        pageSize,
+        pageIndex
+      )
+      if (!error) {
+        setItems(data.data)
```

Comment:
When setting items, you also have to set the new totalItemsCount from the `data.meta`. This is because maybe this API call might have a new filter or new search and the total items changed.